store/commitment/store.go

```diff
@@ -71,49 +78,142 @@ func (c *CommitStore) WorkingStoreInfos(version uint64) []store.StoreInfo {
 		})
 	}
 
-	return storeInfos
+	return &store.CommitInfo{
+		Version:    version,
+		StoreInfos: storeInfos,
+	}
 }
 
 func (c *CommitStore) GetLatestVersion() (uint64, error) {
-	latestVersion := uint64(0)
-	for storeKey, tree := range c.multiTrees {
-		version := tree.GetLatestVersion()
-		if latestVersion != 0 && version != latestVersion {
-			return 0, fmt.Errorf("store %s has version %d, not equal to latest version %d", storeKey, version, latestVersion)
-		}
-		latestVersion = version
+	value, err := c.db.Get([]byte(latestVersionKey))
+	if err != nil {
+		return 0, err
+	}
+	if value == nil {
+		return 0, nil
+	}
+
+	version, _, err := store.DecodeUvarint(value)
+	if err != nil {
+		return 0, err
 	}
 
-	return latestVersion, nil
+	return version, nil
 }
 
 func (c *CommitStore) LoadVersion(targetVersion uint64) error {
+	// Rollback the metadata to the target version.
+	latestVersion, err := c.GetLatestVersion()
+	if err != nil {
+		return err
+	}
+	if targetVersion < latestVersion {
+		batch := c.db.NewBatch()
+		for version := latestVersion; version > targetVersion; version-- {
+			cInfoKey := []byte(fmt.Sprintf(commitInfoKeyFmt, version))
+			if err := batch.Delete(cInfoKey); err != nil {
+				return err
+			}
+		}
+		if err := batch.WriteSync(); err != nil {
+			return err
+		}
+	}
+
 	for _, tree := range c.multiTrees {
 		if err := tree.LoadVersion(targetVersion); err != nil {
 			return err
 		}
 	}
 
-	return nil
+	var cInfo *store.CommitInfo
+	if targetVersion > latestVersion {
+		cInfo = c.WorkingCommitInfo(targetVersion)
+	}
```

Comment:
Can you explain when target would be larger than latest? Also, if we call `LoadVersion(v)` where `v <= latestVersion`, we'll set commit info as nil.

Reply:
yes, it is used in snapshot restoring. If commit info is nil, it means rollback and will not update the commit info metadata